daikon-spring/daikon-spring-reactive/daikon-spring-reactive-auth/src/main/java/org/talend/daikon/spring/reactive/sat/authentication/TalendJwtConverter.java

```diff
@@ -21,10 +21,10 @@ public class TalendJwtConverter implements Converter<Jwt, Collection<GrantedAuth
 
     public static final String TALEND_SAT_PERMISSIONS_FIELD = "https://talend.cloud/permissions";
 
+    @SuppressWarnings("unchecked")
     public Collection<GrantedAuthority> convert(Jwt jwt) {
         List<String> ents = Arrays.asList(((String) jwt.getClaims().getOrDefault(TALEND_ENTITLEMENTS_FIELD, "")).split(","));
-        List<String> permsList = ((JSONArray) jwt.getClaims().getOrDefault(TALEND_PERMISSIONS_FIELD, new JSONArray())).stream()
-                .map(Object::toString).toList();
+        List<String> permsList = (List<String>) jwt.getClaims().getOrDefault(TALEND_PERMISSIONS_FIELD, Collections.emptyList());
         List<String> satPerms = ((JSONArray) jwt.getClaims().getOrDefault(TALEND_SAT_PERMISSIONS_FIELD, new JSONArray())).stream()
                 .map(Object::toString).toList();
```

Comment:
```suggestion
        List<String> satPerms = (List<String>) jwt.getClaims().getOrDefault(TALEND_SAT_PERMISSIONS_FIELD, Collections.emptyList());
```
Shouldn't we apply the same change for SAT permissions?

Reply:
i was wondering about it too

Reply:
done